prov/efa/src/rxr/rxr_atomic.c

```diff
@@ -156,12 +156,12 @@ ssize_t rxr_atomic_generic_efa(struct rxr_ep *rxr_ep,
 		 * the information whether the peer
 		 * support it or not.
 		 */
-		err = rxr_pkt_wait_handshake(rxr_ep, tx_entry->addr, peer);
+		err = rxr_pkt_trigger_handshake(rxr_ep, tx_entry->addr, peer);
 		if (OFI_UNLIKELY(err))
 			goto out;
 
-		assert(peer->flags & RXR_PEER_HANDSHAKE_RECEIVED);
-		if (!rxr_peer_support_delivery_complete(peer)) {
+		if (peer->flags & RXR_PEER_HANDSHAKE_RECEIVED &&
```

Comment:
Can we move ```peer->flags & RXR_PEER_HANDSHAKE_RECEIVED``` to inside ```rxr_peer_support_delivery_complete```?

Reply:
`rxr_peer_support_delivery_complete` returns true when both of the following two statements are true:

1. `peer->flags & RXR_PEER_HANDSHAKE_RECEIVED`
2. `peer->features[0] & RXR_REQ_FEATURE_DELIVERY_COMPLETE`

So if `rxr_peer_support_delivery_complete` returns `false`. It means that the peer might not support `FI_DELIVERY_COMPLETE`. We want to return `-FI_EOPNOTSUPP` here when we know for sure that the peer does not support `FI_DELIVERY_COMPLETE`

Reply:
Currently sender will keep returning `FI_EAGAIN` when the sender do not know if the peer supports `FI_DELIVERY_COMPLETE`.